extensions/access_log_policy/plugin.cc

```diff
@@ -143,6 +147,26 @@ void PluginContext::onLog() {
   setFilterStateValue(false);
 }
 
+bool PluginContext::isRequestFailed() {
+  // Check if HTTP request is a failure.
+  int64_t http_response_code = 0;
+  getValue({kResponse, kCode}, &http_response_code);
+  // If request is a failure, log it.
+  if (http_response_code != 200) {
```

Comment:
Not related to this pr, but should this check 2xx? Also what about 1xx and 3xx?

Reply:
Every thing other than 200 is treated as an error and hence we will log that request.